Python/ceval_macros.h

```diff
@@ -449,3 +449,38 @@ do { \
 #define EXIT_TO_TRACE() goto exit_to_trace
 #define EXIT_TO_TIER1() goto exit_to_tier1
 #define EXIT_TO_TIER1_DYNAMIC() goto exit_to_tier1_dynamic;
+
+/* Stackref macros */
+
+/* How much scratch space to give stackref to PyObject* conversion. */
+#ifdef __APPLE__
+/* macOS seems to have less stack space on non-main threads*/
+#define MAX_STACKREF_SCRATCH 0
```

Comment:
This is the source of your macOS bug. Arrays of size 0 are undefined behavior.

Reply:
:facepalm: I forgot about that. Thank you!